lib/acts_as_ar_query.rb

```diff
@@ -194,6 +194,35 @@ def append_hash_arg(symbol, *val)
     end
   end
 
+  def append_hash_array_arg(symbol, default, *val)
+    val = val.flatten.compact
+    val = val.first if val.size == 1 && val.first.kind_of?(Hash)
```

Comment:
Would it make sense to use [`extract_options!`](http://api.rubyonrails.org/classes/Array.html#method-i-extract_options-21) here?  I think that's what you are trying to accomplish.